addons/binding/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/handler/MieleBridgeHandler.java

```diff
@@ -0,0 +1,585 @@
+        connection.setDoOutput(true);
+        connection.connect();
+
+        OutputStream out = null;
+
+        try {
+            out = connection.getOutputStream();
+
+            out.write(data.getBytes());
+            out.flush();
+            out.close();
+
+            int statusCode = connection.getResponseCode();
+            if (statusCode != HttpURLConnection.HTTP_OK) {
+                logger.error("An unexpected status code was returned : '{}'", statusCode);
+            }
+        } finally {
+            if (out != null) {
+                out.close();
+            }
+        }
+
+        String responseEncoding = connection.getHeaderField("Content-Encoding");
+        responseEncoding = (responseEncoding == null ? "" : responseEncoding.trim());
+
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+
+        InputStream in = connection.getInputStream();
+        try {
+            in = connection.getInputStream();
+            if ("gzip".equalsIgnoreCase(responseEncoding)) {
+                in = new GZIPInputStream(in);
+            }
+            in = new BufferedInputStream(in);
+
+            byte[] buff = new byte[1024];
+            int n;
+            while ((n = in.read(buff)) > 0) {
+                bos.write(buff, 0, n);
+            }
+            bos.flush();
+            bos.close();
+        } finally {
+            if (in != null) {
+                in.close();
+            }
+        }
+
+        return bos.toString();
+    }
+
+    private synchronized void onUpdate() {
+        logger.debug("Scheduling the miele polling job");
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            logger.trace("Scheduling the miele polling job period is {}", POLLING_PERIOD);
+            pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 0, POLLING_PERIOD, TimeUnit.SECONDS);
+            logger.trace("Scheduling the miele polling job Job is done ?{}", pollingJob.isDone());
+        }
+        logger.debug("Scheduling the miele event listener job");
+
+        if (eventListenerJob == null || eventListenerJob.isCancelled()) {
+            eventListenerJob = scheduler.schedule(eventListenerRunnable, 0, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * This method is called whenever the connection to the given {@link MieleBridge} is lost.
+     *
+     */
+    public void onConnectionLost() {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.BRIDGE_OFFLINE);
+    }
+
+    /**
+     * This method is called whenever the connection to the given {@link MieleBridge} is resumed.
+     *
+     * @param bridge the hue bridge the connection is resumed to
+     */
+    public void onConnectionResumed() {
+        updateStatus(ThingStatus.ONLINE);
+        for (Thing thing : getThing().getThings()) {
+            ThingHandler handler = thing.getHandler();
+            if (handler != null) {
+                handler.initialize();
```

Comment:
initialize() should really only be called by the framework; if it contains code that should be called upon handler initialization as well as when Things become ONLINE again, better move that into a separate method.